### PR TITLE
#2835 amenity-recycling handling

### PR DESCRIPTION
**Pull request by shaforostoff**

### CODE REVIEW

**Comment by tatiana-yan** [comment]
@kgalushka is ok with this PR and she added strings to translate to her translation backlog.

The only problem is: if some container has multiple types we'll see only one of them in place page(PP): if container has amenity-recycling-glass_bottles and amenity-recycling-paper and amenity-recycling-plastic we'll have only one of this types in PP.

We agreed with @kgalushka to add all recycling types to subtitle as we do it for cuisines:
Screenshot_20200520_115305
I'll make PR.

File: data/mapcss-mapping.csv [hunk 1]
```diff
@@ -106,7 +106,7 @@ landuse|cemetery;105;
 sport|soccer;106;
 boundary|administrative|4;[boundary=administrative][admin_level=4];;name;int_name;107;
 amenity|cafe;108;
-amenity|recycling;109;
+amenity|recycling;[amenity=recycling][recycling_type=centre];;name;int_name;109;
```
Comment (tatiana-yan) [inline data/mapcss-mapping.csv:109]: now we will lose all recycling containers with no recycling type specified 

I think we need to introduce amenity|recycling|container type with rules
`[amenity=recycling][recycling_type=container],[amenity=recycling][!recycling_type]`

Comment (shaforostoff) [reply data/mapcss-mapping.csv:109]: I see. So with the plan to use approach similar to cuisines, recycling types should be a separate category. I have prepared another PR assisting that:
https://github.com/mapsme/omim/pull/13124/

This approach will require much more changes to the code, so I hope you will it up ;)

Comment (tatiana-yan) [reply data/mapcss-mapping.csv:109]: I was going to do things similar to #13124 (I thought it's easier to do by myself than to explain but maybe I was wrong). I think it's better to merge #13124 (except data/categories_recycling.txt) with first commit of this request and generate resources/strings again for second and third commits.

Comment (shaforostoff) [reply data/mapcss-mapping.csv:109]: заменил batteries на toxic (обычно вместе с батарейками принимают люминесцентные лампочки, иногда ещё и градусники) и добавил scrap metal

переводы не генерировал - не смог победить ruby. а для генерирования drules мне пришлось заменить python на python2 в generate_drules.sh и вызывать его на маке, но ес-но в коммит я это не добавлял

Comment (tatiana-yan) [reply data/mapcss-mapping.csv:109]: почти у всех разработчиков мак и почти у всех каталина (надо собирать ios), у тех кому нужен generate_drules просто два питоновских venv, но надо бы как-нибудь переписать его на python3

strings -- ок, сгенерила и допушила в реквест

File: data/categories.txt [hunk 2]
```diff
@@ -3667,7 +3667,7 @@ fa:زباله دانی
 
 amenity-recycling
 en:Recycling Center|U+1F6AE|U+267B
-ru:Приём вторсырья|утиль|сырье|сбор сырья|стеклотара|макулатура|прием металлов
+ru:Приём вторсырья|утиль|сырье|сбор сырья
```
Comment (mpimenov) [inline data/categories.txt:3670]: s/сырье/сырьё/

File: data/categories.txt [hunk 3]
```diff
@@ -3698,6 +3698,68 @@ sk:Zberné suroviny
 sw:Chombo cha kutia taka
 fa:مرکز بازیافت
 
+amenity-recycling_container
+en:Recycling container|U+1F6AE|U+267B
+es:Contenedor de reciclaje
+de:Recyclingbehälter
+ru:Контейнер для вторсырья|утиль
```
Comment (mpimenov) [inline data/categories.txt:3705]: en, ru, then others (copy the order from an older category)

File: data/categories.txt [hunk 4]
```diff
@@ -3698,6 +3698,68 @@ sk:Zberné suroviny
...
+uk:Контейнер для вторсировини
+
+recycling-toxic
+en:Toxic waste|Betteries
```
Comment (mpimenov) [inline data/categories.txt:3709]: s/Betteries/Batteries/

File: data/strings/types_strings.txt [hunk 5]
```diff
@@ -2517,6 +2517,68 @@
     sk = Zberné suroviny
     fa = مرکز بازیافت
 
+  [type.amenity.recycling_container]
+    en = Recycling container
+    es = Contenedor de reciclaje
+    de = Recyclingbehälter
+    ru = Контейнер для вторсырья
```
Comment (mpimenov) [inline data/strings/types_strings.txt:2524]: copy the language order from an older translation, please

Comment (shaforostoff) [reply data/strings/types_strings.txt:2524]: done

File: data/categories.txt (hunk 4 above)
Comment (shaforostoff) [reply data/categories.txt]: fixed

File: data/categories.txt (hunk 3 above)
Comment (shaforostoff) [reply data/categories.txt:3703]: done

File: data/categories.txt (hunk 2 above)
Comment (shaforostoff) [reply data/categories.txt]: fixed

File: data/strings/types_strings.txt (hunk 5 above)
Comment (mpimenov) [reply data/strings/types_strings.txt:2524]: ping on this comment

Comment (shaforostoff) [reply data/strings/types_strings.txt:2522]: sorry, somehow I missed this. fixed now

File: data/mapcss-mapping.csv (hunk 1 above)
Comment (shaforostoff) [reply data/mapcss-mapping.csv:109]: спасибо!

**Pull request merged**